cvxpy/tests/solver_test_helpers.py

```diff
@@ -173,45 +174,45 @@ def check_stationary_lagrangian(self, places) -> None:
                     diag_entries = np.diag(opt_var.value)
-                    corrected_bad_norm = np.linalg.norm(diag_entries) / np.sqrt(opt_var.size)
+                    dual_cone_violation = np.linalg.norm(diag_entries) / np.sqrt(opt_var.size)
                     if diag_entries > 10**(-places):
-                        bad_norms.append((corrected_bad_norm, opt_var))
+                        bad_norms.append((dual_cone_violation, opt_var))
                 elif opt_var.is_psd():
                     """The PSD cone is self-dual"""
                     g_bad_mat = cp.Constant(np.reshape(v.data, opt_var.shape))
                     tmp_con = g_bad_mat >> 0
-                    corrected_bad_norm = tmp_con.residual
-                    if corrected_bad_norm > 10**(-places):
-                        bad_norms.append((corrected_bad_norm, opt_var))
+                    dual_cone_violation = tmp_con.residual
+                    if dual_cone_violation > 10**(-places):
+                        bad_norms.append((dual_cone_violation, opt_var))
                 elif opt_var.is_nsd():
                     """The NSD cone is also self-dual"""
                     g_bad_mat = cp.Constant(np.reshape(v.data, opt_var.shape))
                     tmp_con = g_bad_mat << 0
-                    corrected_bad_norm = tmp_con.residual
-                    if corrected_bad_norm > 10**(-places):
-                        bad_norms.append((corrected_bad_norm, opt_var))
+                    dual_cone_violation = tmp_con.residual
+                    if dual_cone_violation > 10**(-places):
+                        bad_norms.append((dual_cone_violation, opt_var))
                 elif opt_var.is_nonpos():
                     """The cone of matrices with all entries nonpos is self-dual"""
                     g_bad_mat = cp.Constant(np.reshape(v.data, opt_var.shape))
                     tmp_con = g_bad_mat <= 0
-                    corrected_bad_norm = np.linalg.norm(g_bad_mat.residual) / np.sqrt(opt_var.size)
-                    if corrected_bad_norm > 10**(-places):
-                        bad_norms.append((corrected_bad_norm, opt_var))
+                    dual_cone_violation = np.linalg.norm(g_bad_mat.residual) / np.sqrt(opt_var.size)
+                    if dual_cone_violation > 10**(-places):
+                        bad_norms.append((dual_cone_violation, opt_var))
                 elif opt_var.is_nonneg():
                     """The cone of matrices with all entries nonneg is self-dual"""
                     g_bad_mat = cp.Constant(np.reshape(v.data, opt_var.shape))
                     tmp_con = g_bad_mat >= 0
-                    corrected_bad_norm = np.linalg.norm(g_bad_mat.residual) / np.sqrt(opt_var.size)
-                    if corrected_bad_norm > 10**(-places):
-                        bad_norms.append((corrected_bad_norm, opt_var))
+                    dual_cone_violation = np.linalg.norm(g_bad_mat.residual) / np.sqrt(opt_var.size)
```

Comment:
You're trying to compute ``Constant.residual`` here when you should be trying to compute ``Constraint.residual``. That should raise an error. If tests pass with the code as-is then that suggests to me that tests aren't hitting this code path.

Reply:
Right! I meant to write `tmp_con.residual` there :)

And yes, there are currently no tests for flag variables (except the SDP ones which incorporate `symmetric=True`) --- which is why I was considering adding a new `StandardTestXXX` class within `test_KKT.py` which included tests for testing out this this `if-else` ladder

Reply:
@aryamanjeendgar, yes, please add such tests. A note: for any explicit constraints you use, there's no need to check dual variables beyond their domains. Complementarity, stationarity, primal feasibility, and feasibility w.r.t. dual domains are sufficient for a correctness check.